### DISCUSSION TITLE
Remove the requirement to repeat the name of the section in the end of the section

### DISCUSSION THREAD

**Discussion by spullara** [mustache/spec#175] [Ideas]
This is especially relevant as we look at filters which may become very verbose. I'll start the conversation with a proposal where any end tag could be replaced with {{/}}. For example, today:
```
{{#section}}
Some text
{{/section}}
```
Can optionally be:
```
{{#section}}
Some text
{{/}}
```

**Comment by jgonggrijp** [comment]
I am not against it.

I mean, I don't look forward to lots of users completely abandoning the practice of explicit closing tags and only writing `{{/}}` everywhere. But I guess this can be somewhat reduced by upholding the practice (at least for short keys) in documentation. Maybe that could be a recommendation (#164).

Other than that, I only see reasons to allow it, even if not very strong:

- It does not complicate parsing, at least not much.
- It is true that closing tags can get verbose with dotted names and filters.
- I didn't count, but my impression is that a lot of implementations already allow empty closing tags.

It would not break anything for end users. It would be a breaking change for implementers, though only by limiting the extent to which they can pass the specs. I think it should still be an optional extension for that reason, or be postponed to 2.0.

Questions:

How about `{{#foo.bar.baz}}{{/foo}}`? I feel it looks too much like a mismatching pair.

How about `{{# someList | parent | onlyOdd | enumerate }}{{/ long pipe chain}}`? That seems more reasonable to me.

Then again, `{{#section}}{{/look, I used a closing tag for some irrelevant comment}}` does not seem desirable.

How about parents and blocks? Blocks have no reason to get very long, but parents could potentially get long with dynamic names.

**Comment by agentgt** [comment]
I have similar thoughts as @jgonggrijp . One thing that is currently nice about Mustache is there is not a lot of multiple ways to do things with syntactical sugar.

Its actually one of the things I despise about YAML (as well as handlebars on the templating side) is that there are many multiple ways to do things compared to say JSON.

You could argue the dotted path is sugar but it actually has different meaning than nesting sections.

The other is based on my experience of using SEXP like for HTML generation is that it gets very confusing even with proper indentation as often the start tag is very far away.

In fact I often see HTML designers put in things like

```html
<!-- navigation -->
<div class="nav">
....
</div>
<!-- /navigation -->
```

The reason is that even `</div>` is not providing enough context.

But Mustache is being used more for stuff other than HTML so I can see some value in it where the templating content is less verbose like HTML.

**Reply by agentgt** [reply]
Also I think allowing `{{/}}` cannot be an optional module as it breaks the syntax of the language.

Per the spec guidelines in the readme.md:

```
Optional Modules

It does not affect the core syntax of the language.
```

Filters get a hall pass because the tags name content was never limited other than I believe newline.

That is regardless of filters:

```
{{#blah|stuff}}
...
{{/blah|stuff}}
```

Is syntactically legal mustache.
